app/scripts/metamask-controller.js

```diff
@@ -407,14 +409,16 @@ module.exports = class MetamaskController extends EventEmitter {
 //=============================================================================
 
   /**
-   * Creates a new Vault(?) and create a new keychain(?)
+   * Creates a new Vault and create a new keychain.
    *
-   * A vault is ...
+   * A vault, or KeyringController, is a controller that contains
+   * many different account strategies, currently called Keyrings.
+   * Creating it new means wiping all previous keyrings.
    *
-   * A keychain is ...
+   * A keychain, or keyring, controls many accounts with a single backup and signing strategy.
+   * For example, a mnemonic phrase can generate many accounts, and is a keyring.
    *
-   *
-   * @param  {} password
+   * @param  {string} password
    *
    * @returns {} vault
```

Comment:
It may be a good idea to explicitly type this to have an any type (`*`) or an Object type (`Object`). Some parsers may not handle a complete type omission gracefully. This comment applies throughout.

Reply:
I totally missed this one